fix missing `<cstdint>` include

**Describe the bug**
`vt` fails to build with `clang 16.0.6` on Fedora 38:
```
/vt/src/vt/configs/error/stack_out.h:54:44: error: use of undeclared identifier 'uint64_t'
using StackTupleType  = std::tuple<int32_t,uint64_t,std::string,std::size_t>;
                                           ^
```

This is because of [changes in Standard Library headers in GCC 13](https://gcc.gnu.org/gcc-13/porting_to.html) (see `Header dependency changes`).

--- a/src/vt/configs/error/stack_out.h
+++ b/src/vt/configs/error/stack_out.h
@@ -44,6 +44,7 @@
 #if !defined INCLUDED_VT_CONFIGS_ERROR_STACK_OUT_H
 #define INCLUDED_VT_CONFIGS_ERROR_STACK_OUT_H
 
+#include <cstdint>
 #include <cstdlib>
 #include <string>
 #include <tuple>

--- a/src/vt/utils/compress/decompressor_base.h
+++ b/src/vt/utils/compress/decompressor_base.h
@@ -45,6 +45,7 @@
 #define INCLUDED_VT_UTILS_COMPRESS_DECOMPRESSOR_BASE_H
 
 #include <string>
+#include <cstdint>
 #include <cstdlib>
 
 namespace vt { namespace util { namespace compress {

